src/extension.ts

```diff
@@ -146,20 +142,74 @@ export async function activate(context: vscode.ExtensionContext) {
     apiSecretDisposable,
     apiCredentialsTestDisposable,
     diffDisposable,
-    commentDisposable,
     sidebarDisposable,
     vscode.window.onDidChangeActiveTextEditor(
       async () => await updateTaskbar()
     ),
     vscode.window.onDidChangeActiveTextEditor(
-      async () => await loadFileComments()
+      async () => await loadFileDecorator()
     ),
     exportCommentsFileDisposable,
     exportCommentsFolderDisposable,
     vscode.window.registerFileDecorationProvider(fileDecorator),
-    deleteCommentDisposable,
     assayUpdaterDisposable
   );
+
+  // Comment API
+  const commentController = vscode.comments.createCommentController(
+    "assay-comments",
+    "Assay"
+  );
+
+  // Fetch & restore the current workspace's comments from cache.
+  // Ensure fetches are complete before allowing commentController to be visible.
+  await fetchCommentsFromCache(commentController).then(() => {
```

Comment:
This line is doing something a bit weird: it's both awaiting the resolution of the promise returned by `fetchFilesFromCache()` and using `.then()` on that promise to perform additional processing on the result. We should remove the `await` here in order to avoid blocking  the rest of the initialization of the extension on completion of this work.